ibtk/include/ibtk/ibtk_utilities.h

```diff
@@ -61,55 +61,12 @@ IBTK_ENABLE_EXTRA_WARNINGS
 #if __has_cpp_attribute(deprecated)
 #define IBTK_DEPRECATED(msg) [[deprecated(msg)]]
 #else
-#define IBTK_DEPRECATED(msg)
+#define IBTK_DEPRECATED(msg) [[deprecated]]
```

Comment:
I am sure that I am being dumb, but if `__has_cpp_attribute(deprecated)` is false, shouldn't this be:
```suggestion
#define IBTK_DEPRECATED(msg)
```
?

Reply:
Huh - looks like that should be the case. It looks like we don't test any compiler new enough to have `__has_cpp_attribute` but not C++14 so it isn't an issue. Now would be a good time to update to C++14.

We've had this in-place for a year and nothing bad has happened so I'm disinclined to change it.

Reply:
If we bump up to C++14 can we just use `[[deprecated]]` everywhere instead of using our own macro?

In fact, we might as well consider updating to C++17 since -- AFAIK -- that is now required by libMesh.

Reply:
C++17 would be nice - most people now have at least GCC 9 so it isn't a particularly onerous requirement.

We started requiring C++11 in 2018 so requiring C++17 in 2024 is a logical progression.